openbanking/deploy/setup-ce.sh

```diff
@@ -0,0 +1,1040 @@
+            "type": "string",
+            "allow_weak_decoding": false
+        },
+        {
+            "source": "given_name",
+            "target": "given_name",
+            "type": "string",
+            "allow_weak_decoding": false
+        },
+        {
+            "source": "family_name",
+            "target": "family_name",
+            "type": "string",
+            "allow_weak_decoding": false
+        },
+        {
+            "source": "email",
+            "target": "email",
+            "type": "string",
+            "allow_weak_decoding": false
+        },
+        {
+            "source": "email_verified",
+            "target": "email_verified",
+            "type": "bool",
+            "allow_weak_decoding": false
+        },
+        {
+            "source": "phone_number",
+            "target": "phone_number",
+            "type": "string",
+            "allow_weak_decoding": false
+        },
+        {
+            "source": "phone_number_verified",
+            "target": "phone_number_verified",
+            "type": "bool",
+            "allow_weak_decoding": false
+        },
+        {
+            "source": "customer_id",
+            "target": "customer_id",
+            "type": "string",
+            "allow_weak_decoding": false
+        }
+    ],
+    "static_amr": [],
+    "transformer": {
+        "enabled": false,
+        "script": ""
+    },
+    "config": {
+        "enable_stateful_ctx": false,
+        "stateful_ctx_duration": "0s"
+    },
+    "discovery_settings": {
+        "domains": [],
+        "instant_redirect": false
+    },
+    "token_exchange_settings": {
+        "enabled": false
+    },
+    "identity_pool_id": null,
+    "settings": {
+        "hint": false
+    },
+    "credentials": {
+        "users": [
+            {
+                "username": "user",
+                "password": "p@ssw0rd!",
+                "email": "",
+                "email_verified": false,
+                "phone_number": "",
+                "phone_number_verified": false,
+                "authentication_context": {
+                    "name": "user"
+                },
+                "additional_attributes": {
+                    "customer_id": "bfb689fb-7745-45b9-bbaa-b21e00072447"
+                }
+            }
+        ]
+    }
+}'
+}
+
+function update_consent_app_url {
+    echo "Updating consent app url with arg $1"
+
+    curl --request PUT \
+  --url https://$DOMAIN/api/admin/$TENANT_ID/servers/cdr-import-demo/server-consent \
+  --header "Authorization: Bearer $ACCESS_TOKEN" \
+  --header 'Content-Type: application/json' \
+  --data '{
+    "tenant_id": "ce-apigeex-demo",
+    "server_id": "cdr-import-demo",
+    "client_id": "cdr-import-demo-cah778mkor34hf1058s0",
+    "type": "custom",
+    "oidc": {},
+    "custom": {
+        "server_consent_url": '"\"$1\""'
+    },
+    "openbanking": {
+        "bank_url": ""
+    }
+}'
+}
+
+function update_financroo_redirect_url {
```

Comment:
Is it possible to refactor `update_financroo_redirect_url` and `create_client_financroo` to use the same code with the exception of parameterizing the redirect url? If someone wanted to change the financroo configuration in `create_client_financroo` (e.g. different grant_types), and then ran `update_financroo_redirect_url` it would revert that change.

Reply:
The `update_financroo_redirect_url` is used because after we create the workspace we then run a different script for apigeex. Once the apigeex script finishes it then produces the redirect url we want which we then update using the aforementioned function. So for the demo this value isn't determined by this script or a user independently.